lib/rules/require-await.js

```diff
@@ -77,7 +107,11 @@ module.exports = {
                         name: capitalizeFirstLetter(
                             astUtils.getFunctionNameWithKind(node)
                         )
-                    }
+                    },
+                    suggest: asyncRange ? [{
+                        messageId: "removeAsync",
+                        fix: fixer => fixer.replaceTextRange(asyncRange, addSemiColon ? ";" : "")
+                    }] : []
```

Comment:
I'm not sure if `asyncRange` can ever be falsy. At least I think this should never happen in JavaScript. If we want to keep this ternary expression, maybe it would be better to move the logic that calculates `nextToken` and `addSemicolon` into the fixer function, so that they're not calculated unnecessarily when no suggestion is provided. But I'd be also fine with assuming that `asyncRange` is always an object and dropping the check. Thoughts?

Reply:
Ah, good spotting. The ternary was left over from when the suggestion was not provided because ASI was in use. Now that we add a semicolon when ASI is in use, the ternary is no longer needed. I'll remove it.